packages/hash/api/src/model/entityType.model.ts

```diff
@@ -234,13 +234,25 @@ class __EntityType {
     return dbEntityTypes.map((entityType) => new EntityType(entityType));
   }
 
-  static async getEntityTypeParents(
-    client: DBClient,
-    params: { entityTypeId: string },
-  ) {
-    const dbEntityTypes = await client.getEntityTypeParents(params);
+  async getParentEntityTypes(client: DBClient) {
+    const parentSchema$ids = (
+      Array.isArray(this.properties?.allOf) ? this.properties.allOf : []
+    )
+      .filter(
+        (allOfEntry: any): allOfEntry is { $ref: string } =>
+          "$ref" in allOfEntry,
+      )
+      .map((allOfEntry) => allOfEntry.$ref);
+
+    const parentEntityTypes = await Promise.all(
+      parentSchema$ids.map((schema$id) =>
+        client.getEntityTypeBySchema$id({ schema$id }),
```

Comment:
When removing the existing `get parents` SQL query, I realized it wasn't worth the hassle to implement a query that does this (`getEntityTypeBySchema$id`) with a list of `schema$id`. I opted to follow what we discussed as an alternative @benwerner01